config/environments/test.rb

```diff
@@ -57,8 +54,15 @@
   config.x.default_host = 'lvh.me'
   config.x.client_port = 3200
   config.x.server_port = 7979
+  config.x.default_app_host = "#{config.x.default_host}:3200"
   config.x.default_user_password = 'lolololol'
 
+  # the value should be 'lvh.me:3200'
+  config.x.initial_host = config.x.default_app_host
+
+  # We will assume that we are running on xxx.lvh.me:3200, where xxx is the tenant
+  config.action_mailer.default_url_options = { host: config.x.default_app_host }
```

Comment:
Why is your `default_url_options` in test.rb referencing `default_app_host` but in development.rb it is
`config.action_mailer.default_url_options = { host: config.x.default_host }`?

Reply:
technically, the value of the host inside `default_url_option` should always be pointing towards `lvh.me:{port}` to make it consistent with the `development` environment.

However, the setup inside `development` and `test` for `default_app_host` and `default_host` are switched (inside `development`, it's stored inside `default_host` while inside `test`, it's stored inside `default_app_host`).

I think to make it consistent naming-wise, I'll refer to the `config.x.initial_host` for the `host` value inside `default_url_options`.